apis/v1alpha1/httproute_types.go

```diff
@@ -267,15 +264,18 @@ type HTTPHeaderMatch struct {
 	// Please read the implementation's documentation to determine the supported
 	// dialect.
 	//
+	// HTTP Header names MUST be always matched case-insensitively (RFC 2616 -
+	// section 4.2).
+	//
 	// +kubebuilder:default=Exact
 	Type HeaderMatchType `json:"type,omitempty"`
 
 	// Values is a map of HTTP Headers to be matched.
 	// It MUST contain at least one entry.
 	//
 	// The HTTP header field name to match is the map key, and the
-	// value of the HTTP header is the map value. HTTP header field
-	// names MUST be matched case-insensitively.
+	// value of the HTTP header is the map value. HTTP header field names MUST
+	// be always matched case-insensitively.
```

Comment:
Nit: I think `always` is unnecessary here.

Reply:
Similar comment as above about some kind of alternative wording for `case-insensitively`.